src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -144,6 +146,25 @@ public void verify() throws StartupException
             test.execute();
     }
 
+    public static final StartupCheck checkPreviousVersion = () -> {
+        String previousVersionString = FBUtilities.getPreviousReleaseVersionString();
+        if (null != previousVersionString)
+        {
+            CassandraVersion previousVersion = new CassandraVersion(previousVersionString);
+            boolean incompatible = previousVersion.major < 3;
+
+            if (previousVersion.major == 3)
+            {
+                incompatible |= previousVersion.minor == 0 && previousVersion.patch < 25;
+                incompatible |= previousVersion.minor == 11 && previousVersion.patch < 11;
+                incompatible |= previousVersion.minor != 0 && previousVersion.minor != 11;
+            }
+            if (incompatible)
+                throw new StartupException(StartupException.ERR_WRONG_DISK_STATE,
+                        String.format("Only upgrades from >=3.0.25 and >=3.11.11 are supported (and sstables >= `me` format), found %s", previousVersionString));
```

Comment:
todo: need to check what DSE 5.1 reports as its value here.

Reply:
dse 5.1 will report 3.11.4 which is problematic. this logic is also problematic if an operator is doing an offline upgrade with system tables data in-situ.